src/lab16.py

```diff
@@ -340,6 +381,10 @@ def layout(self):
                 self.children.set(self.temp_children)
                 self.temp_children = None
 
+        height_dependencies = [child.height for child in self.children.get()]
```

Comment:
Ok, I am on board with this general approach but I'm a little unhappy with the `get` call here. In my head, I was always thinking about `get` as an unsafe command and `read` as a safe one. (Because `get` allows you to not set up the dependency.) So I was always auditing my code by looking for each `read` call and making sure they are only in `paint` or similar phases. Probably not fixable.

Reply:
Updated to not use `get()`